[12.0][IMP] purchase_sale_inter_company: make move deletion overridable

RicardCForgeFlow opened the pull request:
After https://github.com/OCA/multi-company/pull/508

We can experience slow performance when confirming intercompany orders

## Proposal
Do not unlink the original stock move of the purchase order to improve action_confirm performance

RicardCForgeFlow committed:
[IMP] purchase_sale_inter_company: make move deletion overridable

This will enable, for example, the use of async deletion
via queue_job

MiquelRForgeFlow commented on purchase_sale_inter_company/models/sale_order.py:
```diff
@@ -45,6 +45,10 @@ def assert_intercompany_prices_equal(self):
                 _('Error. The following lines do not match on'
                   ' the remote order: %s') % "\n".join(unequal_line))
 
+    @api.multi
```
```suggestion
    @api.model
```

MiquelRForgeFlow commented:
<img width="142" height="71" alt="Selection_5200" src="https://github.com/user-attachments/assets/14fb4bae-c6ef-4a44-9234-eae057eb37c6" />

HviorForgeFlow committed:
[IMP] purchase_sale_inter_company: extract split logic into _sync_split_purchase_picking

Extract the N-pickings-to-1-receipt split block from action_confirm into a
dedicated _sync_split_purchase_picking method so subclasses can override the
split strategy without duplicating the full action_confirm.

No behaviour change — the extracted method is identical to the inlined block.
